File: Backend/DTO/UserDto.cs
namespace Backend.DTO;

public class UserDto
{
    public required string Id { get; init; }
    public required string Firstname { get; set; }
    public required string Lastname { get; set; }
    public required string Email { get; set; }
}

public class CreateUserDto
{
    public required string Firstname { get; set; }
    public required string Lastname { get; set; }
    public required string Email { get; set; }
    public required string Password { get; set; }
}

public class UpdatedUserDto : CreateUserDto
{
    public required string Id { get; init; }
}

public class ConnectUserDto
{
    public required string Email { get; set; }
    public required string Password { get; set; }
}

File: Frontend/src/Dto/UserDto.ts
export class UserDto {
    id: number
    firstname: string
    lastname: string
    pseudo: string
    role: "Admin" | "User"

    constructor(id: number, firstname: string, lastname:string, pseudo: string, role: 'Admin' | 'User'){
        this.id = id;
        this.firstname = firstname;
        this.lastname = lastname;
        this.pseudo = pseudo;
        this.role = role;
    }
}

export class CreateUserDto {
    firstname: string
    lastname: string
    pseudo: string
    email: string
    password: string


    constructor(firstname: string, lastname: string, pseudo: string, email: string, password: string) {
        this.firstname = firstname;
        this.lastname = lastname;
        this.pseudo = pseudo;
        this.email = email;
        this.password = password;
    }
}